src/Selector/Xpath/Manipulator.php

```diff
@@ -66,4 +66,50 @@ public function prepend($xpath, $prefix)
 
         return implode(' | ', $expressions);
     }
+
+    /**
+     * Splits the XPath into parts that are separated by the union operator.
+     *
+     * @param string $xpath
+     *
+     * @return string[]
+     */
+    private function splitUnionParts($xpath)
+    {
+        // Split any unions into individual expressions. We need to iterate
+        // through the string to correctly parse opening/closing quotes and
+        // braces which is not possible with regular expressions.
+        $unionParts = array();
+        $singleQuoteOpen = false;
+        $doubleQuoteOpen = false;
+        $bracketOpen = 0;
+        $lastUnion = 0;
+        for ($i = 0; $i < strlen($xpath); $i++) {
+            if ($xpath{$i} === "'" && $doubleQuoteOpen === false) {
+                if ($singleQuoteOpen === true) {
+                    $singleQuoteOpen = false;
+                } else {
+                    $singleQuoteOpen = true;
+                }
+            } else if ($xpath{$i} === '"' && $singleQuoteOpen === false) {
+                if ($doubleQuoteOpen === true) {
+                    $doubleQuoteOpen = false;
+                } else {
+                    $doubleQuoteOpen = true;
+                }
+            } else if ($singleQuoteOpen === false && $doubleQuoteOpen === false) {
+                if ($xpath{$i} === '[') {
```

Comment:
please use ``$xpath[$i]`` rather than ``$xpath{$i}`` to access the element.

And I think defining a ``$char`` variable may be more readable.